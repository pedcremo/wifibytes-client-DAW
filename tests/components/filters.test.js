import Filters from '../../src/components/catalog/filters';
import filtros from '../json_endpoints/filtros.json';
import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme from 'enzyme';
import { shallow } from "enzyme";
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

beforeEach(() => {
    // Set up our document body
    document.body.innerHTML =
        `<div id="main" class="container-fluid pl-0 pr-0">
        </div>`;
});

it('Filters component renders the Filters correctly', () => {
    const rendered = renderer.create(<Filters filters={filtros} />);
    expect(rendered.toJSON()).toMatchSnapshot();
});

it("Filters render must be called and it works properly", () => {
    const filters = Enzyme.shallow(<Filters filters={filtros} />);
    expect(filters).toHaveLength(1);
});

it("Check if there are exactly 5 filters", () => {
    let filters = shallow(<Filters filters={filtros} />);
    expect(filters.find("select.form-control").length).toBe(5);
});

it("Check if the dropdowns are not empty", () => {
    let filters = shallow(<Filters filters={filtros} />);
    const texts = filters.find('label').map(node => node.text());
    expect(texts).toEqual(['Screen:', 'CPU:', 'RAM:', 'Brand:', 'Camera:']);
});

it("Check if there exist the a tag with the class 'btn-secondary' has 'Buy' text", () => {
    let filters = shallow(<Filters filters={filtros} />);
    const texts = filters.find('select').map(node => node.text());
    expect(texts).not.toBeNull();
});
