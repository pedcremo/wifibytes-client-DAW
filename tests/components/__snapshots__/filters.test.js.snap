// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters component renders the Filters correctly 1`] = `
<div
  className="row"
>
  <div
    className="form-group"
  >
    <label
      htmlFor="{key}"
    >
      Screen
      :
    </label>
    <select
      className="form-control"
      id="{key}"
      onChange={[Function]}
    >
      <option>
        3.5
      </option>
      <option>
        4
      </option>
      <option>
        4.2
      </option>
      <option>
        4.5
      </option>
      <option>
        5
      </option>
      <option>
        5.2
      </option>
      <option>
        10.1
      </option>
      <option>
        6
      </option>
      <option>
        5.5
      </option>
    </select>
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="{key}"
    >
      CPU
      :
    </label>
    <select
      className="form-control"
      id="{key}"
      onChange={[Function]}
    >
      <option>
        Core 2.5GHz
      </option>
      <option>
        Quad-Core 1.3GHz
      </option>
      <option>
        Quad-Core 2.5GHz
      </option>
      <option>
        Octa-Core 1.5GHz
      </option>
      <option>
        Quad-Core 1.2GHz
      </option>
      <option>
        Accesorio
      </option>
      <option>
        Octa Core,2.2 Ghz
      </option>
      <option>
        Quad-Core 1 GHz
      </option>
      <option>
        Octa-Core 2GHz
      </option>
      <option>
        Deca-Core 2.1GHz
      </option>
    </select>
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="{key}"
    >
      RAM
      :
    </label>
    <select
      className="form-control"
      id="{key}"
      onChange={[Function]}
    >
      <option>
        512mb
      </option>
      <option>
        1Gb
      </option>
      <option>
        2Gb
      </option>
      <option>
        4Gb
      </option>
      <option>
        8Gb
      </option>
      <option>
        Accesorio
      </option>
      <option>
        3 Gb
      </option>
    </select>
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="{key}"
    >
      Brand
      :
    </label>
    <select
      className="form-control"
      id="{key}"
      onChange={[Function]}
    >
      <option>
        Nokia
      </option>
      <option>
        Apple
      </option>
      <option>
        Sony
      </option>
      <option>
        LG
      </option>
      <option>
        Huawei
      </option>
      <option>
        Samsung
      </option>
      <option>
        bq
      </option>
      <option>
        Hannspree
      </option>
      <option>
        Universal
      </option>
      <option>
        Xiaomi
      </option>
      <option>
        Wiko
      </option>
    </select>
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="{key}"
    >
      Camera
      :
    </label>
    <select
      className="form-control"
      id="{key}"
      onChange={[Function]}
    >
      <option>
        8
      </option>
      <option>
        10.2
      </option>
      <option>
        12.5
      </option>
      <option>
        13
      </option>
      <option>
        0
      </option>
      <option>
        12
      </option>
      <option>
        16
      </option>
      <option>
        5
      </option>
    </select>
  </div>
</div>
`;
